modules/UI/videolayout/RemoteVideo.js

```diff
@@ -133,92 +156,64 @@ RemoteVideo.prototype._isHovered = function () {
-            muteClassName = 'mutelink';
+            onRemoteControlToggle = this._stopRemoteControl;
+            remoteControlState = REMOTE_CONTROL_MENU_STATES.STARTED;
         }
-
-        let muteHandler = this._muteHandler.bind(this);
-        let kickHandler = this._kickHandler.bind(this);
-
-        menuItems = [
-            {
-                id: 'mutelink_' + this.id,
-                handler: muteHandler,
-                icon: 'icon-mic-disabled',
-                className: muteClassName,
-                data: {
-                    i18n: muteTranslationKey
-                }
-            }, {
-                id: 'ejectlink_' + this.id,
-                handler: kickHandler,
-                icon: 'icon-kick',
-                data: {
-                    i18n: 'videothumbnail.kick'
-                }
-            }
-        ];
     }
 
-    if(this._supportsRemoteControl) {
-        let icon, handler, className;
-        if(APP.remoteControl.controller.getRequestedParticipant()
-            === this.id) {
-            handler = () => {};
-            className = "requestRemoteControlLink disabled";
-            icon = "remote-control-spinner fa fa-spinner fa-spin";
-        } else if(!APP.remoteControl.controller.isStarted()) {
-            handler = this._requestRemoteControlPermissions.bind(this);
-            icon = "fa fa-play";
-            className = "requestRemoteControlLink";
-        } else {
-            handler = this._stopRemoteControl.bind(this);
-            icon = "fa fa-stop";
-            className = "requestRemoteControlLink";
-        }
-        menuItems.push({
-            id: 'remoteControl_' + this.id,
-            handler,
-            icon,
-            className,
-            data: {
-                i18n: 'videothumbnail.remoteControl'
-            }
-        });
-    }
-
-    menuItems.forEach(el => {
-        let menuItem = this._generatePopupMenuItem(el);
-        popupmenuElement.appendChild(menuItem);
-    });
+    let initialVolumeValue, onVolumeChange;
 
-    // feature check for volume setting as temasys objects cannot adjust volume
+    // Feature check for volume setting as temasys objects cannot adjust volume.
     if (this._canSetAudioVolume()) {
-        const volumeScale = 100;
-        const volumeSlider = this._generatePopupMenuSliderItem({
-            handler: this._setAudioVolume.bind(this, volumeScale),
-            icon: 'icon-volume',
-            initialValue: this._getAudioElement().volume * volumeScale,
-            maxValue: volumeScale
-        });
-        popupmenuElement.appendChild(volumeSlider);
+        initialVolumeValue = this._getAudioElement().volume;
+        onVolumeChange = this._setAudioVolume;
     }
 
-    APP.translation.translateElement($(popupmenuElement));
-
-    return popupmenuElement;
+    const { isModerator } = APP.conference;
+    const participantID = this.id;
+
+    /* jshint ignore:start */
+    return (
+        <ul
```

Comment:
Right now all the callbacks are still handled in RemoteVideo so I would be passing those callbacks into RemoteMenu and then into the buttons. How about as soon as my other work in progress lands I try moving those callbacks into actions? The buttons become way more encapsulated.